### ISSUE TITLE
Usage within React Components

### ISSUE THREAD

**Issue by churcho** [arillo/meteor-flow-router-helpers#24]
Could you please provide an example of how I can use this package within a React component.

**Comment by banglashi** [comment]
Hi @churcho 
I haven't had timt o play with React yet, so no Idea if the helper is usable out of the box. I will let you know as soon as I do.

**Comment by Ricemast** [comment]
:+1:

**Comment by zimme** [comment]
zimme/meteor-active-route#45 this is where I'm tracking react support for [`zimme:active-route`](https://atmospherejs.com/zimme/active-route), which is implied by this package.

Should we create mixins? I've heard that mixins is going away in react with es2015 support?
I have a javascript api, `ActiveRoute.name`/`ActiveRoute.path`, maybe this is enough to use
it in react?

Is using the javascript api directly in react is an option, most of the helper functionality in this package
could just use the underlying functions directly, or wrap then in a neat namespace.

I've seen something about using `Blaze.render`/`Blaze.renderWithData` in react to be able to use blaze templates, maybe that works for helpers too?

**Comment by Ricemast** [comment]
[This](https://github.com/themeteorchef/getting-started-with-react/blob/master/code/client/helpers/flow-router.js) is currently what I am using. Simply functions wrapped in a global object. I'm not sure if that is any help?

**Comment by zimme** [comment]
@Ricemast I haven't look into React much at all, is it enough to just have a javascript api available like that `FlowHelpers.helper()` or is there a need to do tighter integration in some way? because if the javascript api thing is enough then I think this work can be done quickly with this package and I think I just need to add some options to my package to have it return the class instead of true when enabled.

**Comment by Ricemast** [comment]
Yes. A simple JS API would work just fine :) You can render the result of any JS function in React components. Here is how I am currently using the API from my last comment :: `<a href={FlowHelpers.pathFor('addUser')} className="btn btn-success">Add User</a>`

Edit: Maybe checkout the [blog post](https://themeteorchef.com/recipes/getting-started-with-react/#tmc-updating-our-flow-router-helpers) I learnt this from a few hours ago!

Edit 2: `href={}` shouldn't have had quotes around the `{}`

**Comment by zimme** [comment]
@Ricemast Great, currently `ActiveRoute.name`/`ActiveRoute.path` returns true or false. I'll look into adding an option to have those return the `class`/`className` string when the `react` package is available with an option to turn this behaviour off.

For this package it seems like a no brainer to just have all the helper functions exported to blaze helpers exported through a javascript object which the blaze helpers uses. I like the namespace `FlowHelpers` but if any1 have a better suggestion for the namespace feel free to leave suggestions.

I'll try and fix my package this weekend and see if I can't put together a PR for this one.

Currently working on es2015 rewrites of my packages so I might end up getting lost in that, if so ping me during the week and I'll try and schedule some time for this.

**Comment by Ricemast** [comment]
@zimme That's great :smile: I like `FlowHelpers` too. Good luck with your rewrites!

**Comment by banglashi** [comment]
@zimme anything I can help you with, let me know.
